`fmtDurationSigned` overflow panic when given `std.math.minInt(i64)`

### Zig Version

0.14.0

### Steps to Reproduce and Observed Behavior

Running the following:

```zig
const std = @import("std");

pub fn main() void {
    std.debug.print("{}\n", .{ std.fmt.fmtDurationSigned(std.math.minInt(i64)) });
}
```

Causes this panic:
```
panic: integer overflow
path/to/zig/lib/std/fmt.zig:1384:9: 0x7ff66d744d3f in formatDurationSigned__anon_26883
        const data = FormatDurationData{ .ns = @as(u64, @intCast(-ns)), .negative = true };
        ^
```

### Expected Behavior

I would expect this to not overflow. 

The `@as(u64, @intCast(-ns))` should just be replaced with `@abs(ns)` and the problem is solved.

--- a/lib/std/fmt.zig
+++ b/lib/std/fmt.zig
@@ -1380,13 +1380,8 @@ test fmtDuration {
 }
 
 fn formatDurationSigned(ns: i64, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void {
-    if (ns < 0) {
-        const data = FormatDurationData{ .ns = @as(u64, @intCast(-ns)), .negative = true };
-        try formatDuration(data, fmt, options, writer);
-    } else {
-        const data = FormatDurationData{ .ns = @as(u64, @intCast(ns)) };
-        try formatDuration(data, fmt, options, writer);
-    }
+    const data = FormatDurationData{ .ns = @abs(ns), .negative = ns < 0 };
+    try formatDuration(data, fmt, options, writer);
 }
 
 /// Return a Formatter for number of nanoseconds according to its signed magnitude:
@@ -1457,6 +1452,7 @@ test fmtDurationSigned {
         .{ .s = "-1y1m999ns", .d = -(365 * std.time.ns_per_day + std.time.ns_per_min + 999) },
         .{ .s = "292y24w3d23h47m16.854s", .d = math.maxInt(i64) },
         .{ .s = "-292y24w3d23h47m16.854s", .d = math.minInt(i64) + 1 },
+        .{ .s = "-292y24w3d23h47m16.854s", .d = math.minInt(i64) },
     }) |tc| {
         const slice = try bufPrint(&buf, "{}", .{fmtDurationSigned(tc.d)});
         try std.testing.expectEqualStrings(tc.s, slice);

